rest-api-spec/src/main/resources/rest-api-spec/test/index/90_flat_object.yml

```diff
@@ -0,0 +1,438 @@
+---
+# Create flat_object mapping
+setup:
+  - do:
+      indices.create:
+        index: test
+        body:
+          mappings:
+            properties:
+              ISBN13:
+                type : "keyword"
+              catalog:
+                type : "flat_object"
+              required_matches:
+                type : "long"
+
+# Delete Index when connection is teardown
+---
+teardown:
+  - do:
+      indices.delete:
+        index: test
+
+# Upload a sample document to index
+---
+"Flat object test":
+  - skip:
+      version: " - 2.99.99"
+      reason: "flat_object is introduced in 3.0.0 in main branch"
+  - do:
+      index:
+        index: test
+        id:    1
+        body:  {
+          "ISBN13": "V9781933988177",
+          "catalog": {
+            "title": "Lucene in Action",
+            "author":
+              {
+                "surname": "McCandless",
+                "given": "Mike"
```

Comment:
How the usage of flat objects look from scripts perspective? Can we access it from painless / expressions? We need to add at least a few tests to `SearchFieldsIT` (or better place) to demonstrate that.

Reply:
Added tests in SearchFieldsIT https://github.com/opensearch-project/OpenSearch/blob/d8b3dd3a1d7f76bd1f7477ad3494be866e2ea3fd/server/src/internalClusterTest/java/org/opensearch/search/fields/SearchFieldsIT.java